shared/utility/vision/Vision.cpp

```diff
@@ -43,7 +43,7 @@ namespace utility::vision {
         ofs.close();
     }
 
-    const auto getSubImage(uint x, uint y, uint width, uint height, const std::vector<uint8_t>& data) {
+    auto getSubImage(uint x, uint y, uint width, uint height, const std::vector<uint8_t>& data) {
```

Comment:
Doesn't this remove the constness of the data?

Also, should we pass the ints by reference?

Reply:
having const on a return type that is not a reference is meaninglesss

Reply:
We return a eigen map, which does have an internal reference to the data.

Reply:
The map is mapping a `const Eigen::Matrix` though, so I think that it should still be const where it needs to be

Reply:
We're mapping a `std::vector<uint8_t>`?

Reply:
I am reading this code as it creates an `const Eigen::Matrix` from the data from the vector, then returns a submatrix (`Eigen::Map`) of that `const Eigen::Matrix`
It's not a hash map. It's basically a submatrix view. [eigen map docs](https://eigen.tuxfamily.org/dox/classEigen_1_1Map.html)
```c++
        return Eigen::Map<const Eigen::Matrix<uint8_t, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>>(data.data(),
                                                                                                         height,
                                                                                                         width)
            .block<5, 5>(y - 2, x - 2);
```

Trent's point is a thing too. It doesn't do anything, and I am pretty sure that's what the clang-tidy warning was looking at.

Reply:
Basically, it does nothing because it would return a const version of whatever you return. Then if you assigned it to a variable it'd just copy it into the new variable (because it has to copy over the const thing). Therefore it does nothing